xskillscore/core/comparative.py

```diff
@@ -0,0 +1,151 @@
+import warnings
+
+import scipy.stats as st
+import xarray as xr
+
+
+def sign_test(
+    observation,
+    forecast1,
+    forecast2,
+    dim=None,
+    alpha=0.05,
+    metric=None,
+    orientation='negative',
+):
+    """
+        Returns the Delsole and Tippett sign test over the given time dimension.
+
+        Parameters
+        ----------
+        observation : xarray.Dataset or xarray.DataArray or None
+            observation to be compared to both forecasts.
+            If ``None``, then assume that forecast1 and forecast2 have already been
+            compared to observation. If metric is None, this assumes forecasts to be
+            already compared to observation before and ignores observation. Please
+            adjust ``orientation`` accordingly.
+        forecast1 : xarray.Dataset or xarray.DataArray
+            forecast1 to be compared to observation
+        forecast2 : xarray.Dataset or xarray.DataArray
+            forecast2 to be compared to observation
+        dim : str
+            time dimension of dimension over which to compute the random walk.
+            This dimension is not reduced, unlike in other xskillscore functions.
+        alpha : float
+            significance level for random walk.
+        metric : callable, optional
+            metric to compare forecast# with observation if metric is not None. If
```

Comment:
typo with # here and below. Also add `str` to types next to `callable` since they can be converted. Might be nice to clearly state that this is an `xskillscore` metric, e.g. `pearson_r`.

Reply:
forecast# is for forecast1 and forecast2

Reply:
Could make it "metric to compare forecast N ..." or "the given forecast to observations ..." but I'm fine with it as is.